simtools/job_submission/job_manager.py

```diff
@@ -0,0 +1,207 @@
+            self._logger.error(
+                "Requested workflow manager not found: {}".format(
+                    self.submitCommand))
+            raise
+
+    def test_submission_system(self):
+        """
+        Check that the requested workload manager exist on the
+        system this script is executed
+
+        Raises
+        ------
+        MissingWorkloadManager
+            if workflow manager is not found
+
+        """
+
+        if self.submitCommand is None:
+            return
+        elif self.submitCommand.find("qsub") >= 0:
+            if gen.program_is_executable('qsub'):
+                return
+            raise MissingWorkloadManager
+        elif self.submitCommand.find("condor_submit") >= 0:
+            if gen.program_is_executable('condor_submit'):
+                return
+            raise MissingWorkloadManager
+        elif self.submitCommand.find("seriell_script") >= 0:
+            return
+        elif self.submitCommand.find("more") >= 0:
+            return
+
+        raise MissingWorkloadManager
+
+    def submit(
+        self,
+        run_script=None,
+        run_out_file=None,
+    ):
+        """
+        Submit a job described by a shell script
+
+        Parameters
+        ----------
+        run_script: string
+            Shell script descring the job to be submitted
+        run_out_file: string
+            Redirect output/error/job stream to this file (out,err,job suffix)
+
+        """
+        self.run_script = str(run_script)
+        self.run_out_file = str(run_out_file)
+
+        self._logger.info(
+            'Submitting script {}'.format(self.run_script))
+        self._logger.info(
+            'Job output stream {}'.format(self.run_out_file+".out"))
+        self._logger.info(
+            'Job error stream {}'.format(self.run_out_file+".err"))
+        self._logger.info(
+            'Job log stream {}'.format(self.run_out_file+".job"))
+
+        if self.submitCommand.find("qsub") >= 0:
+            self._submit_gridengine()
+        elif self.submitCommand.find("condor_submit") >= 0:
+            self._submit_HTcondor()
+        elif self.submitCommand.find("seriell_script") >= 0:
+            self._submit_seriell_script()
+        elif self.submitCommand.find("more") >= 0:
+            self._submit_for_testing()
+
+    def _submit_for_testing(self):
+        """
+        Print submit script to screen for testing
+
+        """
+
+        self._logger.info('Testing script')
+
+        shellCommand = 'more ' + self.run_script
+
+        if not self.test:
+            os.system(shellCommand)
+        else:
+            self._logger.info('Testing (submit_for_testing)')
+
+    def _submit_seriell_script(self):
+        """
+        Run a job script on the command line
+        (no submission to a workload manager)
+
+        """
+
+        self._logger.info('Running script locally')
+
+        shellCommand = self.run_script + \
+            " > " + self.run_out_file + ".out"
```

Comment:
Don't we need here to direct the error stream to the ``.err`` file as well? I am not sure about the log (something I always wanted to handle internally actually).

Reply:
Agree - forgot the error stream. 

What exactly do you want to mean  with `I am not sure about the log...`?

Reply:
When submitting to the batch farm we also set a log file to which the logging goes. When running locally, the log goes to stdout. Not sure if through a unix command we can pipe the log (not the stdout) to a log file. 

I know it's possible in the logging package of python, which is something I have been wanting to do for a while now.